contracts/CommunityVault.sol

```diff
@@ -77,7 +160,21 @@ contract CommunityVault is CommunityOwnable {
         }
 
         for (uint256 i = 0; i < recipients.length; i++) {
+            bool removed = erc721TokenIds[token].remove(tokenIds[i]);
+            if (!removed) {
+                revert CommunityVault_ERC721TokenNotDeposited();
+            }
+
             IERC721(token).safeTransferFrom(address(this), recipients[i], tokenIds[i]);
         }
     }
+
+    /**
+     * @dev Handles the receipt of an ERC721 token.
+     * @return bytes4 Returns `bytes4(keccak256("onERC721Received(address,address,uint256,bytes)"))`
+     *          to indicate the contract implements `onERC721Received` as per ERC721.
+     */
+    function onERC721Received(address, address, uint256, bytes calldata) public pure override returns (bytes4) {
+        return this.onERC721Received.selector;
```

Comment:
I think here we still need to check if:

a) the received `ERC721` doesn't already exist in the set and
b) do a) only if we know for sure that this `onReceived` event wasn't triggered from a previous `depositERC721()` call

Thoughts @0xb337r007 ?

Reply:
I completely skipped the check here on purpose to support only the deposit function. otherwise we might have strange scenarios in case we call `deposit` with a token that doesn't call the `onERC721Received`

Reply:
Hm.. okay maybe we can revisit this in the future.
I might be missing something here, but if such a token would not call `onERC721Received` it's perfectly fine no? We still track via `depositERC721`. I'd even consider this one of the "simpler" cases as it'd just be as if `onERC721Received` didn't exist.

For now we can leave it like this, but I still think we should consider supporting both unless there's good reasons not to.